docs/en-us/containerrecreaterequest.md

```diff
@@ -0,0 +1,108 @@
+---
+title: ContainerRecreateRequest
+---
+# ContainerRecreateRequest
+
+**FEATURE STATE:** Kruise v0.9.0
+
+ContainerRecreateRequest provides a way to let users **restart/recreate** one or more containers in an existing Pod.
+
+Just like the in-place update provided in Kruise, during a container being recreating, the other containers in the same Pod are still running.
+After the recreation completed, nothing changes in the Pod except the container's restartCount will be increased.
+So the files written into the **rootfs** of the previous container will be lost, but the data in volume mounts will keep existing.
+
+Note that this feature relies on `kruise-daemon` to stop the container in Pod.
+So if the `KruiseDaemon` feature-gate is closed, ContainerRecreateRequest will also be disabled.
+
+## Usage
+
+### Submit request
+
+Create a `ContainerRecreateRequest` (short name `CRR`) for each Pod container recreation:
+
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: ContainerRecreateRequest
+metadata:
+  namespace: pod-namespace
+  name: xxx
+spec:
+  podName: pod-name
+  containers:       # list of container name that need to be recreated, at least one
+  - name: app
+  - name: sidecar
+  strategy:
+    failurePolicy: Fail                 # or 'Ignore'. If 'Fail', the CRR will finish when one container failed to stop or recreate
+    orderedRecreate: false              # 'true' indicates to recreate the next container only if the previous one has recreated completely
+    terminationGracePeriodSeconds: 30   # the optional duration in seconds to wait the container terminating gracefully
+    unreadyGracePeriodSeconds: 3        # it is the duration in seconds to mark Pod as not ready over this duration before executing preStop hook and stopping the container
```

Comment:
the duration a container is marked as not ready before its preStop hook is executed and it is stopped